chatbot_handler.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import time
+import re
+
+from abstract_handler import AbstracHandler
+from message_handler import MessageHandler
+
+
+class ChatBotHandler(AbstracHandler):
+
+    def __init__(self, botId, botName, client, socket_delay=1):
+        self.botId = botId
+        self.botName = botName
+        self.mention = self.getMention(botId)
+        self.client = client
+        self.delay = socket_delay
+        self.messageHandler = MessageHandler(self.mention)
+
+    def isPrivate(self, event):
+        """Checks if private slack channel"""
+        return event.get('channel').startswith('D')
+
+    def _isForMe(self, event):
+        eventType = event.get('type')
+
+        if not eventType or eventType != 'message' or \
+                event.get('user') == self.botId:
+            return False
+
+        if self.isPrivate(event):
+            return 'private'
+
+        messageText = event.get('text')
+
+        if re.search(str(self.mention), str(messageText)):
+            return 'mention'
+
+        if re.search(str(self.botName), str(messageText.lower())):
+            return 'name'
+
+    def _postMessage(self, messageText, channel):
+        self.client.api_call(
+            'chat.postMessage', channel=channel,
+            text=messageText, as_user=True)
+
+    def handle(self, message, user, channel, mention):
+        privateChannel = channel.startswith('D')
+        response = self.messageHandler.handle(
+            message, user, mention, privateChannel)
+        if response:
```

Comment:
fico, così è perfetto per gestire un fallback in caso di errore nel parsing del messaggio!